elements/rh-pagination/rh-pagination.ts

```diff
@@ -123,32 +129,44 @@ export class RhPagination extends LitElement {
     const nextHref = this.#nextLink?.href;
     const lastHref = this.#currentLink === this.#lastLink ? undefined : this.#lastLink?.href;
     const currentPage = this.#currentPage.toString();
+    const numericContent = html`
```

Comment:
I would consider breaking this out into its own function, this block should only render if compact is set.

```js
  #numericContent(currentPage: string, lastHref?: string, ) {
    return html`
      <div id="numeric" part="numeric">
        <span id="go-to-page" class="visually-hidden">
          <slot name="go-to-page">
              Go to page
          </slot>
        </span>
        <input inputmode="numeric"
            required
            min=1 max=${this.#links?.length ?? 1}
            aria-labelledby="go-to-page"
            @change=${this.#onChange}
            @keyup=${this.#onKeyup}
            .value=${currentPage}>
        <slot name="out-of">of</slot>
        <a href=${ifDefined(lastHref)}>${this.#links?.length}</a>
      </div>
    `;
  }
```

Then calling that in your method in `render()`:

```js
  render() {
    const { on = '' } = this;
    const { mobile, size } = this.#screen;
    const { dir } = this.#dir;
    const { label, labelFirst, labelPrevious, labelNext, labelLast } = this;
    const firstHref = this.#currentLink === this.#firstLink ? undefined : this.#firstLink?.href;
    const prevHref = this.#prevLink?.href;
    const nextHref = this.#nextLink?.href;
    const lastHref = this.#currentLink === this.#lastLink ? undefined : this.#lastLink?.href;
    const currentPage = this.#currentPage.toString();

    return html`
      <div id="container" class=${classMap({ mobile, [size as string]: true, [dir]: true, [on]: !!on })}>
        <a id="first" class="stepper" href=${ifDefined(firstHref)} ?inert=${!firstHref} aria-label=${labelFirst}>${L2}</a>
        <a id="prev" class="stepper" href=${ifDefined(prevHref)} ?inert=${!prevHref} aria-label=${labelPrevious}>${L1}</a>

        ${!this.variant?.includes('compact') ?
          html`
            <nav ?hidden=${mobile} ?inert=${mobile} aria-label=${label}>
              <slot></slot>
            </nav>
          `
          : html`
            ${this.#numericContent(currentPage, lastHref)}
          `}
        <a id="next" class="stepper" href=${ifDefined(nextHref)} ?inert=${!nextHref} aria-label=${labelNext}>${L1}</a>
        <a id="last" class="stepper" href=${ifDefined(lastHref)} ?inert=${!lastHref} aria-label=${labelLast}>${L2}</a>
        ${this.variant?.includes('compact') ? html`${this.#numericContent(currentPage, lastHref)}` : html``}
      </div>
    `;
  }
```

Reply:
Updated in [e5a747d](https://github.com/RedHat-UX/red-hat-design-system/pull/1587/commits/e5a747d0f904968ac7fb3888fe049168f1fdb39b). 

Changed two things:

  1. The last `this.variant?...` should be `!this.variant?...` and now is.
  1. the `.visually-hidden` class in the `#numericContent` function needed to be conditionally applied.

I like this approach better than my previous one. Nice suggestion 👏